src/main/java/com/jfrog/ide/idea/configuration/GlobalSettings.java

```diff
@@ -52,11 +52,23 @@ public XrayServerConfigImpl getXrayConfig() {
         return this.xrayConfig;
     }
 
+    // Required for deserialization.
     public void setXrayConfig(XrayServerConfigImpl xrayConfig) {
+        this.xrayConfig.setUrl(xrayConfig.getUrl());
+        this.xrayConfig.setUsername(xrayConfig.getUsername());
+        this.xrayConfig.setCredentials(xrayConfig.getCredentialsFromPasswordSafe());
+        this.xrayConfig.setExcludedPaths(xrayConfig.getExcludedPaths());
+    }
+
+    public void updateConfig(XrayServerConfigImpl xrayConfig) {
+        if (this.xrayConfig.getUrl() != null && !this.xrayConfig.getUrl().equals(xrayConfig.getUrl())) {
+            this.xrayConfig.removeCredentialsFromPasswordSafe();
+        }
         this.xrayConfig.setUrl(xrayConfig.getUrl());
         this.xrayConfig.setUsername(xrayConfig.getUsername());
         this.xrayConfig.setPassword(xrayConfig.getPassword());
         this.xrayConfig.setExcludedPaths(xrayConfig.getExcludedPaths());
+        this.xrayConfig.addCredentialsToPasswordSafe();
     }
```

Comment:
This 2 methods have common code. I suggest to reuse common code.
Please also add Javadoc above each one and explain the purpose of each one.